docs/tutorial/go.rst

```diff
@@ -0,0 +1,360 @@
+.. _build-a-rock-for-a-go-application:
+
+Build a rock for a Go application
+------------------------------------
+
+In this tutorial, we'll create a simple Go application and learn how to
+containerise it in a rock, using Rockcraft's ``go-framework``
+:ref:`extension <go-framework-reference>`.
+
+Setup
+=====
+
+.. include:: /reuse/tutorial/setup.rst
+
+.. note::
+    This tutorial requires the ``latest/edge`` channel of Rockcraft. Use
+    ``sudo snap refresh rockcraft --channel latest/edge`` to get the latest
+    edge version.
+
+Finally, create a new directory for this tutorial and go inside it:
+
+.. code-block:: bash
+
+   mkdir go-hello-world
+   cd go-hello-world
+
+Create the Go application
+==============================
+
+Start by creating the "Hello, world" Go application that will be used for
+this tutorial.
+
+In order to test the Go application locally (before packing it into a rock),
+install ``go``.
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:install-go]
+    :end-before: [docs:install-go-end]
+    :dedent: 2
+
+Initialise the new Go module:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:mod-init-go]
+    :end-before: [docs:mod-init-go-end]
+    :dedent: 2
+
+Create a ``main.go`` file, copy the following text into it and then
+save it:
+
+.. literalinclude:: code/go/main.go
+    :language: go
+
+Build the Go application so it can be run:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:go-build]
+    :end-before: [docs:go-build-end]
+    :dedent: 2
+
+A new binary named ``go-hello-world`` should be in the current
+directory.  This binary is only needed for local testing, as
+Rockcraft will compile the Go application. Run the Go application using
+``./go-hello-world`` to verify that it works.
+
+Test the Go application by using ``curl`` to send a request to the root
```

Comment:
I feel like there's a jump here because 'hello world' code in Go does not typically start up an http server. Maybe some context like "The code in ``main.go`` starts up an HTTP server listening on port ``8000``, so we test it by using ``curl``..."